charts/orchestrator/templates/release-candidate.yaml

```diff
@@ -0,0 +1,115 @@
+{{/*
+This file used for including source resources for RHDH and Sonataflow components from RC builds.
+It requires including a secret to enable access to the image registries that contains the builds.
+See https://github.com/janus-idp/operator/blob/main/.rhdh/docs/installing-ci-builds.adoc
+*/}}
+{{- if .Values.isReleaseCandidate }}
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  labels:
+    operators.openshift.org/catalog: "true"
+  name: brew-registry
+spec:
+  repositoryDigestMirrors:
+    - mirrors:
+        - brew.registry.redhat.io/openshift-serverless-1/logic-rhel8-operator
+      source: registry.redhat.io/openshift-serverless-1/logic-rhel8-operator
+    - mirrors:
+        - brew.registry.redhat.io/rh-osbs/openshift-serverless-1-logic-operator-bundle
+      source: registry-proxy.engineering.redhat.com/rh-osbs/openshift-serverless-1-logic-operator-bundle
+---
+apiVersion: operator.openshift.io/v1alpha1
+kind: ImageContentSourcePolicy
+metadata:
+  name: quay-io
+spec:
+  repositoryDigestMirrors:
+  - mirrors:
+    - quay.io/rhdh/rhdh-operator-bundle
+    source: registry.redhat.io/rhdh/rhdh-operator-bundle
+  - mirrors:
+    - quay.io/rhdh/rhdh-operator-bundle
+    source: registry.stage.redhat.io/rhdh/rhdh-operator-bundle
+  - mirrors:
+    - quay.io/rhdh/rhdh-operator-bundle
+    source: registry-proxy.engineering.redhat.com/rh-osbs/rhdh-rhdh-operator-bundle
+  - mirrors:
+    - quay.io/rhdh/rhdh-rhel9-operator
+    source: registry.redhat.io/rhdh/rhdh-rhel9-operator
+  - mirrors:
+    - quay.io/rhdh/rhdh-rhel9-operator
+    source: registry.stage.redhat.io/rhdh/rhdh-rhel9-operator
+  - mirrors:
+    - quay.io/rhdh/rhdh-rhel9-operator
+    source: registry-proxy.engineering.redhat.com/rh-osbs/rhdh-rhdh-rhel9-operator
+  - mirrors:
+    - quay.io/rhdh/rhdh-hub-rhel9
+    source: registry.redhat.io/rhdh/rhdh-hub-rhel9
+  - mirrors:
+    - quay.io/rhdh/rhdh-hub-rhel9
+    source: registry.stage.redhat.io/rhdh/rhdh-hub-rhel9
+  - mirrors:
+    - quay.io/rhdh/rhdh-hub-rhel9
+    source: registry-proxy.engineering.redhat.com/rh-osbs/rhdh-rhdh-hub-rhel9
+  - mirrors:
+    - quay.io
+    source: registry.redhat.io
+  - mirrors:
+    - quay.io
+    source: registry.stage.redhat.io
+  - mirrors:
+    - quay.io
+    source: registry-proxy.engineering.redhat.com
+  - mirrors:
+    - registry.redhat.io
+    source: registry.stage.redhat.io
+  - mirrors:
+    - registry.stage.redhat.io
+    source: registry-proxy.engineering.redhat.com
+  - mirrors:
+    - registry.redhat.io
+    source: registry-proxy.engineering.redhat.com
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: {{ .Values.sonataFlowOperator.subscription.sourceName }}
+  namespace: openshift-marketplace
+spec:
+  displayName: Sonataflow Operator (Brew)
+  {{- if eq .Values.ocpVersion "v4.13" }}
```

Comment:
I suggest to use a map with static values that contain the image URLs and populate the value here to avoid having to manage this conditional in the future. Check the [dig](https://helm.sh/docs/chart_template_guide/function_list/#dig) function.

```
sonataflow:
  images:
    ocpVersion:
      4.13: brew.registry.redhat.io/rh-osbs/iib:735728
      4.14: brew.registry.redhat.io/rh-osbs/iib:735738
      4.15: brew.registry.redhat.io/rh-osbs/iib:735753
      4.16: brew.registry.redhat.io/rh-osbs/iib:735763
```

And then for instance
```
# Retrieve OCP version from `clusterversion` object
$version := semver $ocpVersion
# Check that the version is within the supported range, else `fail()`
{{ dig "ocpVersion" (printf "%s.%s" $version.Major $version.Minor) .Values.sonataflow.images }}
```

Reply:
Use this snippet to retrieve the OCP major and minor versions:
https://gist.github.com/jordigilh/3bd37e86093593d9e6273b8e7bc72181